packages/backend/src/model/BlockRange.ts

```diff
@@ -0,0 +1,76 @@
+import assert from 'assert'
+import { range } from 'lodash'
+
+import { BlockNumber } from '../peripherals/ethereum/types'
+
+export class BlockRange {
+  /** number of the earliest block in range */
+  public readonly from: BlockNumber
+  /** number of the latest block in range */
+  public readonly to: BlockNumber
+  private readonly hashes: ReadonlyMap<BlockNumber, string>
+
+  constructor(
+    /** a list of blocks in order from oldest to newest */
+    blocks: readonly Block[]
+  )
+  constructor(
+    /** parent block range */
+    range: BlockRange,
```

Comment:
This should probably be either a range or a list of blocks, so that one doesn't have to write `new BlockRange(new BlockRange(blocks), from, to)` if the start and end have unknown hashes